### PR TITLE
Add a configuration object for thrift builder

### PR DESCRIPTION
**Pull request by henryxparker**

### CODE REVIEW

File: twitter-finagle/src/main/scala/com/dwolla/util/async/finagle/ThriftClientConfiguration.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.dwolla.util.async.finagle
+
+import com.twitter.finagle.service.RetryBudget
+import com.twitter.util.Duration
+
+case class ThriftClientConfiguration(
+                                      requestTimeout: Duration = Duration.fromSeconds(30),
+                                      sessionAcquireTimeout: Duration = Duration.fromSeconds(30),
```
Comment (DhruvPhumbhra) [inline twitter-finagle/src/main/scala/com/dwolla/util/async/finagle/ThriftClientConfiguration.scala:8]: _nit_: Let's call this `sessionAcquisitionTimeout`

File: twitter-finagle/src/main/scala-ce3/com/dwolla/util/async/finagle/ThriftClient.scala [hunk 2]
```diff
@@ -16,13 +16,22 @@ object ThriftClient {
...
                                   FK: FunctorK[Alg]): Resource[G, Alg[G]] = {
-      val acquire = Sync[G].delay(Thrift.client.build[Alg[F]](dest)).map(_.asyncMapK[G])
+      val acquire =
+        Sync[G].delay(
+          Thrift
+            .client
+            .withSessionPool.maxSize(config.sessionPoolMax)
+            .withRequestTimeout(config.requestTimeout)
+            .withSession.acquisitionTimeout(config.requestTimeout)
+            .withRetryBudget(config.retryBudget)
```
Comment (DhruvPhumbhra) [inline twitter-finagle/src/main/scala-ce3/com/dwolla/util/async/finagle/ThriftClient.scala:32]: I think we want to be exhaustive here so we should also include a config for `.withTransport.connectTimeout()`

**Comment by DhruvPhumbhra** [comment]
This looks good to me but not sure what the next steps are to get this approved and merged. Don't want to "officially" approve this without some consent from Brian

File: twitter-finagle/src/main/scala/com/dwolla/util/async/finagle/ThriftClientConfiguration.scala (hunk 1 above)
Comment (henryxparker) [reply twitter-finagle/src/main/scala/com/dwolla/util/async/finagle/ThriftClientConfiguration.scala]: Renamed

File: twitter-finagle/src/main/scala-ce3/com/dwolla/util/async/finagle/ThriftClient.scala (hunk 2 above)
Comment (henryxparker) [reply twitter-finagle/src/main/scala-ce3/com/dwolla/util/async/finagle/ThriftClient.scala:33]: Added

**Comment by henryxparker** [comment]
> This looks good to me but not sure what the next steps are to get this approved and merged. Don't want to "officially" approve this without some consent from Brian

Understandable, I would feel similarly if this were a bigger change. I think it's okay in this instance: Configuration does need to exist for the finagle clients in some form since our team has already run up against problems with the defaults and considering @bpholt is not available to review any time soon, I think approval from any maintainer should be sufficient this time.

If @bpholt returns and finds the solution unsatisfactory, I will happily open up another PR.

**Review by DhruvPhumbhra — APPROVED** [review]

**Comment by bpholt** [comment]
This seems pretty reasonable for now. FYI using case classes (and maybe default values for parameters?) in public signatures has bad binary compatibility implications, so it's not ideal, but I don't think we need to worry about it until we add MiMa checking (especially since Finagle doesn't really worry about bincompat either AFAICT)

**Pull request merged**